### PR TITLE
Add RSpec

### PR DESCRIPTION
**Pull request by arincon9**
This Pull Request adds RSpec foundation to the app and tests the models.

### CODE REVIEW

File: config/database.yml.example [hunk 1]
```diff
@@ -1,22 +0,0 @@
-default: &default
```
Comment (sanghun89) [inline config/database.yml.example]: wait don't we need this file?

File: test/controllers/home_controller_test.rb [hunk 2]
```diff
@@ -1,9 +0,0 @@
-require 'test_helper'
-
-class HomeControllerTest < ActionDispatch::IntegrationTest
```
Comment (sanghun89) [inline test/controllers/home_controller_test.rb]: why remove this test???

File: test/application_system_test_case.rb [hunk 3]
```diff
@@ -1,5 +0,0 @@
-require "test_helper"
-
-class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
-  driven_by :selenium, using: :chrome, screen_size: [1400, 1400]
```
Comment (sanghun89) [inline test/application_system_test_case.rb]: never knew rails do selenium end to end testing

**Comment by markdamobi** [comment]
> All is 💯 this one - just a quick question - any reason why we are removing fixtures?? are we to use it for later and just make the file as is needed or we don't need the use for it at all?

I don't know if we really need fixtures. I think it'll be okay with just FactoryBot. I don't see factory bot being installed though.

File: config/database.yml.example (hunk 1 above)
Comment (arincon9) [reply config/database.yml.example]: This was by accident

File: test/controllers/home_controller_test.rb (hunk 2 above)
Comment (arincon9) [reply test/controllers/home_controller_test.rb]: Its a placeholder test from the test folder. We are replacing `test` with `spec`

File: test/application_system_test_case.rb (hunk 3 above)
Comment (arincon9) [reply test/application_system_test_case.rb]: We will be doing specs with RSpec and we can add selenium there.

**Comment by arincon9** [comment]
@markdamobi @sanghun89 I think we will want to add fixtures as we get to them. Fixtures are significantly faster than factories and easier to change once the app grows.

**Comment by sanghun89** [comment]
Okay sounds good - those fixture files were created when running the generator command for models, can we either put a .keep or leave the files so we know where to edit/add fixtures?? Otherwise you'd have to read in to the helper file to see where it attempts to read fixtures

**Comment by arincon9** [comment]
@sanghun89 I know you already approved this P.R. but I updated the user specs to use Fixtures. Now we will have a pattern for using fixtures as recommended.

**Comment by markdamobi** [comment]
> @markdamobi @sanghun89 I think we will want to add fixtures as we get to them. Fixtures are significantly faster than factories and easier to change once the app grows.

@arincon9 Are you saying fixtures are more flexible? I don't think that's true. Can you add FactoryBot to the app too and then we can use either or both or have you concluded that we won't use factories?

**Comment by sanghun89** [comment]
> > @markdamobi @sanghun89 I think we will want to add fixtures as we get to them. Fixtures are significantly faster than factories and easier to change once the app grows.
> 
> @arincon9 Are you saying fixtures are more flexible? I don't think that's true. Can you add FactoryBot to the app too and then we can use either or both or have you concluded that we won't use factories?

I'm not familiar with the fixture vs factorybot holy war - seems pretty similar to me with fixture being the default for ralis

**Comment by arincon9** [comment]
@markdamobi I have added FactoryBot 👍

**Comment by markdamobi** [comment]
> @markdamobi I have added FactoryBot 👍

Alright, cool. I haven't used fixtures in the past, so I'll probably start by using them so I can better understand why you all prefer them.

File: spec/models/user_favorite_organization_spec.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+require "rails_helper"
+
+RSpec.describe UserFavoriteOrganization, type: :model do
+  it { should belong_to(:user) }
+  it { should belong_to(:organization) }
```
Comment (markdamobi) [inline spec/models/user_favorite_organization_spec.rb]: @arincon9 can you add a test for the validation that ensures that a user cannot favorite the same organization twice?
`  validates :user_id, uniqueness: { scope: :organization_id }`

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  subject { users(:bob_user) }
+
+  it { should define_enum_for(:favorite_cause).backed_by_column_of_type(:string) }
+
+  it { should have_many(:user_favorite_organizations).dependent(:destroy) }
+  it { should have_many(:donations).dependent(:destroy) }
+  it do
+    should have_many(:favorite_organizations)
+      .through(:user_favorite_organizations)
+      .source(:organization)
```
Comment (markdamobi) [inline spec/models/user_spec.rb]: I had no idea it was possible to be this specific with the shoulda matchers. This is good 👍

File: spec/models/user_favorite_organization_spec.rb (hunk 4 above)
Comment (arincon9) [reply spec/models/user_favorite_organization_spec.rb]: Oops, must have missed this. Thanks for catching it. The spec has been added.

**Comment by arincon9** [comment]
@markdamobi Comments have been addressed, could I please get another look at this?

**Pull request merged**